File: java/src/dev/csn/Soa.java
package dev.csn;

public class Soa {
    public int[] ids;
    public boolean[] selected;

    Soa(Aor src) {
        ids = new int[src.items.size()];
        selected = new boolean[src.items.size()];

        for (var i = 0; i < ids.length; i++) {
            var item = src.items.get(i);
            ids[i] = item.itemId;
            selected[i] = item.selected;
        }
    }

    public boolean selected(int id) {
        for (int i = 0; i < ids.length; ++i) {
            if (ids[i] == id) {
                return selected[i];
            }
        }
        return false;
    }

}

File: cs/soa-benchmark/soa-benchmark/Soa.cs
namespace soa_benchmark
{
    public class Soa
    {
        public readonly int[] Ids;
        public readonly bool[] SelectedItems;


        public Soa(Aor aor)
        {
            Ids = new int[aor.Size];
            SelectedItems = new bool[aor.Size];
            for (var i = 0; i < aor.Size; ++i)
            {
                var item = aor.Items[i];
                Ids[i] = item.Id;
                SelectedItems[i] = item.Selected;
            }
        }

        public bool Selected(int id)
        {
            var i = 0;
            foreach (var currentId in Ids)
            {
                if (currentId == id)
                {
                    return SelectedItems[i];
                }

                i++;
            }

            return false;
        }
        
    }
}